tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1162,6 +1166,52 @@ DeclResultIdMapper::createShaderRecordBufferNV(const HLSLBufferDecl *decl) {
   return bufferVar;
 }
 
+SpirvVariable *
+DeclResultIdMapper::createShaderRecordBufferEXT(const VarDecl *decl) {
```

Comment:
The only difference between this method and `DeclResultIdMapper::createShaderRecordBufferNV(const VarDecl *decl)` is the `"type.ShaderRecordBufferEXT."` and `ContextUsageKind::ShaderRecordBufferEXT`.

Why don't we change `DeclResultIdMapper::createShaderRecordBufferNV(const VarDecl *decl)` to `DeclResultIdMapper::createShaderRecordBuffer(const VarDecl *decl, ContextUsageKind kind)` and use it for both of `ContextUsageKind::ShaderRecordBufferNV` and `ContextUsageKind::ShaderRecordBufferEXT`?

Reply:
I initially didn't as it would expose the `ContextUsageKind` enumeration and I wasn't sure if that was desirable. Updated the code now to your suggestion and and added an assert to catch invalid enum values being passed.